### ISSUE TITLE
async-node target support

### ISSUE THREAD

**Issue by hardfist** [web-infra-dev/rspack#3298]
### What problem does this feature solve?

webpack use async-node target to test, since we want to align with webpack's test,we should support it

### What does the proposed API of configuration look like?

```js
module.exports = {
 target: 'async-node'
}
```

**Comment by underfin** [comment]
Here already has https://github.com/web-infra-dev/rspack/pull/1878

**Comment by Boshen** [comment]
This currently blocks webpack tests,

```
error[internal]: Unsupported chunk loading type $async-node.
```

**Comment by hardfist** [comment]
landed in #1878
